pyls/rpc_manager.py

```diff
@@ -0,0 +1,171 @@
+# Copyright 2017 Palantir Technologies, Inc.
+import logging
+from uuid import uuid1
+
+from concurrent.futures import ThreadPoolExecutor, Future
+from jsonrpc.base import JSONRPCBaseResponse
+from jsonrpc.jsonrpc1 import JSONRPC10Response
+from jsonrpc.jsonrpc2 import JSONRPC20Response, JSONRPC20Request
+from jsonrpc.exceptions import JSONRPCMethodNotFound, JSONRPCInternalError
+
+log = logging.getLogger(__name__)
+
+RESPONSE_CLASS_MAP = {
+    "1.0": JSONRPC10Response,
+    "2.0": JSONRPC20Response
+}
+
+
+class JSONRPCManager(object):
```

Comment:
I wouldn't personally call either of the managers servers. neither really provide server like functionality to the client, they are mostly just there to manage the sending of messages so that the real server can more easily operate